spec/Configuration/DotenvSpec.hs

```diff
@@ -125,14 +124,27 @@ spec = do
   describe "onMissingFile" $ after_ clearEnvs $ do
     context "when target file is present" $
       it "loading works as usual" $ do
-        void $ onMissingFile (loadFile $ Config ["spec/fixtures/.dotenv"] [] True False) (return [])
+        void $ onMissingFile (loadFile sampleConfig {configOverride = True}) (return [])
         lookupEnv "DOTENV" `shouldReturn` Just "true"
 
     context "when target file is missing" $
       it "executes supplied handler instead" $
-        onMissingFile (True <$ loadFile (Config ["spec/fixtures/foo"] [] True False)) (return False)
+        onMissingFile (True <$ loadFile sampleConfig {configPath= ["spec/fixtures/.foo"], configOverride = True}) (return False)
           `shouldReturn` False
 
+  describe "onDuplicatedKeys" $ after_ clearEnvs $ do
+    context "when target file has duplicated key and the duplicate are forbidden" $
+      it "throws an error" $
+        loadFile sampleConfig { configPath = ["spec/fixtures/.dotenv", "spec/fixtures/.dotenv"], allowDuplicates = False }
+          `shouldThrow` anyIOException
+    context "when target file does not have duplicated key and the duplicate are forbidden" $
+      it "works as usual" $ do
+        void $ loadFile sampleConfig { allowDuplicates = False }
+        lookupEnv "DOTENV" `shouldReturn` Just "true"
```

Comment:
@FranzGB I think we should double check that the first env variable is set in this case.

Reply:
That's exactly what this test case is doing 🙌